src/playlist-maker.js

```diff
@@ -133,217 +122,336 @@ const factory = (player, initialList, initialIndex = 0) => {
    * @return {Array}
    *         The playlist
    */
-  const playlist = player.playlist = function(newList, newIndex = 0) {
+  const playlist = player.playlist = (newList, newIndex = 0) => {
     if (Array.isArray(newList)) {
       list = newList.slice();
       if (newIndex !== -1) {
         playlist.currentItem(newIndex);
       }
-      playlist.changeTimeout_ = window.setTimeout(() => {
-        player.trigger('playlistchange');
-      }, 0);
+      player.setTimeout(() => player.trigger('playlistchange'), 1);
     }
 
     // Always return a shallow clone of the playlist list.
     return list.slice();
   };
 
+  // On a new source, if there is no current item, disable auto-advance.
   player.on('loadstart', () => {
     if (playlist.currentItem() === -1) {
       autoadvance.reset(player);
     }
   });
 
-  player.on('dispose', () => {
-    window.clearTimeout(playlist.changeTimeout_);
-  });
+  playlist.currentIndex_ = -1;
```

Comment:
Switching from the old `assign` to this assignment-based approach removes a largely unnecessary function and results in a slightly smaller minified file size.